transport/controller.cc

```diff
@@ -69,7 +69,9 @@ future<> controller::do_start_server() {
         cql_transport::cql_server_config cql_server_config;
         cql_server_config.timeout_config = make_timeout_config(cfg);
         cql_server_config.max_request_size = service::get_local_storage_service()._service_memory_total;
-        cql_server_config.max_concurrent_requests = cfg.max_concurrent_requests_per_shard;
+        cql_server_config.get_max_concurrent_requests = [ss = std::ref(service::get_storage_service())] () -> utils::updateable_value<uint32_t> {
+            return ss.get().local().db().local().get_config().max_concurrent_requests_per_shard;
+        };
```

Comment:
updateable_value was supposed to be shard-safe exactly for this use-case!

Reply:
```c++
// A T that can be updated at runtime; uses updateable_value_base to track
// the source as the object is moved or copied. Copying across shards is supported.
template <typename T>
class updateable_value : public updateable_value_base {
```

Reply:
Well, the current implementation is not :(

The bug is already on master, so I wanted to provide a simple fix fast. If `updateable_value` was supposed to be shard safe then I agree it should be fixed too, but I think that it will require more effort than this.

Reply:
What about reverting 4b856cf62d839b6d555703f6ba902ff57a132898? We can undo the revert after updateable_value is fixed.

Reply:
If we revert, then we will lose the ability to hot-reload the param. If we apply this PR we can later revert it after updateable_value is fixed.

Reply:
Regardless from what we do with this PR, I'll create a separate issue about updateable_value not being shard-safe.

Reply:
https://github.com/scylladb/scylla/issues/7316

Reply:
Ok. I'll merge.